app/models/message.rb

```diff
@@ -168,109 +99,48 @@ def disable_group_email_link(to_person)
     else
       "\"#{I18n.t('messages.do_not_reply')}\" <#{Site.current.noreply_email}>"
     end
   end
 
-  before_create :generate_security_code
-
-  def generate_security_code
-    begin
-      code = rand(999999)
-      write_attribute :code, code
-    end until code > 0
-  end
-
   def code_hash
     Digest::MD5.hexdigest(code.to_s)[0..5]
   end
 
-  def streamable?
-    person_id and not to_person_id and group
-  end
-
-  after_create :create_as_stream_item
-
-  def create_as_stream_item
-    return unless streamable?
-    StreamItem.create!(
-      title:           subject,
-      body:            html_body.present? ? html_body : body,
-      text:            !html_body.present?,
-      person_id:       person_id,
-      group_id:        group_id,
-      streamable_type: 'Message',
-      streamable_id:   id,
-      created_at:      created_at,
-      shared:          !!group
-    )
-  end
-
-  after_update :update_stream_items
-
-  def update_stream_items
-    return unless streamable?
-    StreamItem.where(streamable_type: "Message", streamable_id: id).each do |stream_item|
-      stream_item.title = subject
-      if html_body.present?
-        stream_item.body = html_body
-        stream_item.text = false
-      else
-        stream_item.body = body
-        stream_item.text = true
-      end
-      stream_item.save
-    end
-  end
-
-  after_destroy :delete_stream_items
-
-  def delete_stream_items
-    StreamItem.destroy_all(streamable_type: 'Message', streamable_id: id)
-  end
-
   def self.preview(attributes)
-    msg = Message.new(attributes)
-    Notifier.full_message(Person.new(email: 'test@example.com'), msg)
+    Notifier.full_message(Person.new(email: 'test@example.com'), Message.new(attributes))
   end
 
   def self.create_with_attachments(attributes, files)
     message = Message.create(attributes.update(dont_send: true))
-    unless message.errors.any?
-      files.select { |f| f && f.size > 0 }.each do |file|
-        attachment = message.attachments.create(
-          name:         File.split(file.original_filename).last,
-          content_type: file.content_type,
-          file:         file
-        )
-        if attachment.errors.any?
-          attachment.errors.full_messages.each { |e| message.errors.add(:base, e) }
-          return message
-        end
-      end
-      message.dont_send = false
-      message.enqueue_send
-    end
-    message
+    message.attach_files(files) unless message.errors.any?
+  end
+
+  def attach_files(files)
+    files.reject { |f| f.size.zero? }.each do |file|
+      attachment = attachments.create(
+        name:         File.split(file.original_filename).last,
+        content_type: file.content_type,
+        file:         file
+      )
+      attachment.errors.full_messages.each { |e| errors.add(:base, e) }
+      return self if errors.any?
+    end
+    self.dont_send = false
+    enqueue_send
+    self
```

Comment:
I don't think the `enqueue_send` should be part of this method -- it should be part of the `create_with_attachments` method. At least, that seems less surprising to me.